### PR TITLE
[improvement](statistics)Add log for estimating hive table row count logic.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Add log for estimating hive table row count logic to help investigate user issues.

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSExternalTable.java [hunk 1]
```diff
@@ -525,7 +525,7 @@ public long fetchRowCount() {
         long rowCount = getRowCountFromExternalSource();
         // Only hive table supports estimate row count by listing file.
         if (rowCount == -1 && dlaType.equals(DLAType.HIVE)) {
-            LOG.debug("Will estimate row count from file list.");
+            LOG.info("Will estimate row count for table {} from file list.", name);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSExternalTable.java:528]: print catalog.database.table?

**Review by morrySnow — APPROVED** [review]

**Comment by Jibing-Li** [comment]
run performance

**Comment by Jibing-Li** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/ExternalRowCountCache.java [hunk 2]
```diff
@@ -113,6 +113,7 @@ public long getCachedRowCount(long catalogId, long dbId, long tableId) {
             if (f.isDone()) {
                 return f.get().orElse(-1L);
             }
+            LOG.info("Row count for table {}.{}.{} is still processing.", catalogId, dbId, tableId);
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/ExternalRowCountCache.java:116]: is this the main execution path? log.info may print too much to kill the performance.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSExternalTable.java [hunk 3]
```diff
@@ -863,14 +865,20 @@ private long getRowCountFromFileList() {
...
 
         int totalPartitionSize = partitionValues == null ? 1 : partitionValues.getIdToPartitionItem().size();
         if (samplePartitionSize != 0 && samplePartitionSize < totalPartitionSize) {
+            LOG.info("Table {} sampled {} of {} partitions, sampled size is {}",
+                    name, samplePartitionSize, totalPartitionSize, totalSize);
             totalSize = totalSize * totalPartitionSize / samplePartitionSize;
         }
-        return totalSize / estimatedRowSize;
+        long rows = totalSize / estimatedRowSize;
+        LOG.info("Table {} rows {}, total size is {}, estimatedRowSize is {}",
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSExternalTable.java:879]: better only to print the info under unhealthy cases, such as "size==0" cases, and avoid printing all.

**Review by xzj7019 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/datasource/ExternalRowCountCache.java (hunk 2 above)
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/ExternalRowCountCache.java:116]: This log will only be printed when the cache is not loaded. Will not print any more after the cache is available.

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSExternalTable.java (hunk 3 above)
Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/hive/HMSExternalTable.java:879]: This function is only called while loading the cache, will not print this log after cache is available.

**Pull request merged**